server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1641,13 +1654,28 @@ public BlobContainer shardContainer(IndexId indexId, int shardId) {
      *
      * @param repositorySettings repository settings
      * @param setting            setting to use to configure rate limiter
-     * @return rate limiter or null of no throttling is needed
+     * @param warnIfOverRecovery log a warning if rate limit setting is over the effective recovery rate limit
+     * @return rate limiter or null if no throttling is needed
      */
-    private static RateLimiter getRateLimiter(Settings repositorySettings, Setting<ByteSizeValue> setting) {
+    private RateLimiter getRateLimiter(Settings repositorySettings, Setting<ByteSizeValue> setting, boolean warnIfOverRecovery) {
         ByteSizeValue maxSnapshotBytesPerSec = setting.get(repositorySettings);
         if (maxSnapshotBytesPerSec.getBytes() <= 0) {
             return null;
         } else {
+            if (warnIfOverRecovery && recoverySettings.rateLimiter() != null) {
+                double effectiveRecoverySpeed = recoverySettings.rateLimiter().getMBPerSec();
+                if (maxSnapshotBytesPerSec.getMbFrac() > effectiveRecoverySpeed) {
+                    logger.warn(
```

Comment:
This warning will only be printed on cluster state updates and when the repository settings have changed, but I don't think it will if the recovery setting is updated.

I also wonder if we should emit a periodic warning too?

Reply:
Thanks a lot for this feedback, I had not thought of it! I updated the PR and now it also logs it when the recovery setting(s) are updated. Please take another look.

In the process, I also realized that a similar warning should be done for the snapshot restore speed in case it's over the recovery speed, so I made a warning for that as well. And re-adapted the relevant test that tests for the logged warning.

I believe a periodic warning here is an overkill. When we initially were talking about this, if I remember correctly, we were even hesitant to add a warning at all, since we would document it. And I think we were fine with a single warning. But feel free to strengthen your voice here if you think we should really do it a periodic warning. Also, I hope with https://github.com/elastic/elasticsearch/issues/91354 we'll provide better/more visibility in the effective values of the various speeds.